### DISCUSSION TITLE
ChebConv and GCNConv returns nan for specific edge weights

### DISCUSSION THREAD

**Discussion by niko4299** [pyg-team/pytorch_geometric#3580] [Q&A]
Hey! 
So in my task I calculate corrcoef with numpy between nodes and use them as edge weights in network and for some values of edge weights my network returns nan values. Is there some specific interval that weights value need to be or ?

My network consist of only ChebConv(3,3,K = 2). My weights are dimension 1x784 because i have 28 nodes so if needed I can provide weights values. Weights values are between [-1,1].

Ty for your time.

**Comment by rusty1s** [comment]
`ChebConv` and (`GCNConv`) apply symmetric normalization, and therefore only support positive edge weights. This is due to the `deg.pow(-0.5)` call which will produce NaNs for negative values. I think we definitely need a way to detect this and to provide a better error message :)

**Comment by SisyphusMountain** [comment]
Hey, I got this error again and spent 2 hours debugging. It would be great if it could fail with a better error message ! Thanks a lot.
